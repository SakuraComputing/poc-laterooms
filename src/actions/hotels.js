export const addHotel = (hotel) => ({
    type: 'ADD_HOTEL',
    hotel
});

export const setHotels = (hotels) => ({
    type: 'SET_HOTELS',
    hotels
});