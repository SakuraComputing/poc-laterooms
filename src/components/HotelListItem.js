import React from 'react';

const HotelListItem = ({ name, starRating, facilities}) => {
    return (
        <div className="hotel-box">
            <h3>Hotel Name: {name}</h3>
            <div>Star Rating: {starRating}</div>
            <div>Facilities Include: {facilities.toString()}</div>
        </div>
    );
}

export default HotelListItem;
