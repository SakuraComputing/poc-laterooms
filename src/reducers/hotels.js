const hotelReducerDefaultState = [];

export default (state = hotelReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_HOTEL':
            return [
                ...state,
                action.hotel
            ];
        default:
            return state;
    }
};