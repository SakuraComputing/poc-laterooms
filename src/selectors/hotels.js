
export default (hotels, { text }) => {
    return hotels.filter((hotel) => {
        const textMatch = hotel.name.toLowerCase().includes(text.toLowerCase());
        console.log('Text Please', textMatch, hotel);
        return textMatch;
    });
};
