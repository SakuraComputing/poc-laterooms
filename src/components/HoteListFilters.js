import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByStarRating } from '../actions/filters';

export class HotelListFilters extends Component {

  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };

 onSortChange = (e) => {
    this.props.sortByStar(e.target.value);
 }

  render() {
    return (
      <div>
          <input
            type="text"
            placeholder="Hotel Name"
            value={this.props.filters.text}
            onChange={this.onTextChange}
            className="input-box"
          />
          <select className="select"
                  value={this.props.filters.sortBy}
                  onChange={this.onSortChange}
          >
            <option value="">Star Rating</option>
            <option value="Asc">Asc</option>
            <option value="Desc">Desc</option>
          </select>
      </div>
    )
  }
};

const mapStateToProps = (state) => ({
    filters: state.filters
});

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByStar: (sortBy) => dispatch(sortByStarRating(sortBy)),
    // sortByStarDesc: () => dispatch(sortByStarRatingDesc()),
});

export default connect(mapStateToProps, mapDispatchToProps)(HotelListFilters);
