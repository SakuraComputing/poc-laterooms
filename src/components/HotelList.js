import React from 'react';
import { connect } from 'react-redux';
import HotelListItem from './HotelListItem';

export const HotelList = (props) => {
    return (
        <div>
            {props.hotels.map((hotel) => {
                return <HotelListItem key={hotel.name} {...hotel} />;
            })}
        </div>
    );
}
const mapStateToProps = (state) => {
    return {
        hotels: state.hotels
    };
}

export default connect(mapStateToProps)(HotelList);
