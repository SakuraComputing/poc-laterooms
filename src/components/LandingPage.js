import React from 'react';
import HotelList from './HotelList';
import '../App.css';

const LandingPage = () => {
    return (
        <div className="container">
            <h1>Late Rooms</h1>
            <HotelList />
        </div>
    );
}
export default LandingPage;
