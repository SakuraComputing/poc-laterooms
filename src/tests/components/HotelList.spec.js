import React from 'react';
import { shallow } from 'enzyme';
import { HotelList } from '../../components/HotelList';
import hotels from '../../fixtures/hotels';

describe('Hotel List Component', () => {
    it('should render correctly', () => {
        const wrapper = shallow(<HotelList hotels={hotels} />);
        expect(wrapper).toMatchSnapshot();
    })
})

