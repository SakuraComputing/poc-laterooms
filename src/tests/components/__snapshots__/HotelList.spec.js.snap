// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hotel List Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HotelList
    hotels={
      Array [
        Object {
          "facilities": Array [
            "car park",
            "pool",
          ],
          "name": "hotelone",
          "starRating": 5,
        },
        Object {
          "facilities": Array [
            "car park",
            "gym",
          ],
          "name": "hoteltwo",
          "starRating": 3,
        },
        Object {
          "facilities": Array [],
          "name": "hotelthree",
          "starRating": 3,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <HotelListItem
          facilities={
            Array [
              "car park",
              "pool",
            ]
          }
          name="hotelone"
          starRating={5}
        />,
        <HotelListItem
          facilities={
            Array [
              "car park",
              "gym",
            ]
          }
          name="hoteltwo"
          starRating={3}
        />,
        <HotelListItem
          facilities={Array []}
          name="hotelthree"
          starRating={3}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "hotelone",
        "nodeType": "function",
        "props": Object {
          "facilities": Array [
            "car park",
            "pool",
          ],
          "name": "hotelone",
          "starRating": 5,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "hoteltwo",
        "nodeType": "function",
        "props": Object {
          "facilities": Array [
            "car park",
            "gym",
          ],
          "name": "hoteltwo",
          "starRating": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "hotelthree",
        "nodeType": "function",
        "props": Object {
          "facilities": Array [],
          "name": "hotelthree",
          "starRating": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <HotelListItem
            facilities={
              Array [
                "car park",
                "pool",
              ]
            }
            name="hotelone"
            starRating={5}
          />,
          <HotelListItem
            facilities={
              Array [
                "car park",
                "gym",
              ]
            }
            name="hoteltwo"
            starRating={3}
          />,
          <HotelListItem
            facilities={Array []}
            name="hotelthree"
            starRating={3}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "hotelone",
          "nodeType": "function",
          "props": Object {
            "facilities": Array [
              "car park",
              "pool",
            ],
            "name": "hotelone",
            "starRating": 5,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "hoteltwo",
          "nodeType": "function",
          "props": Object {
            "facilities": Array [
              "car park",
              "gym",
            ],
            "name": "hoteltwo",
            "starRating": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "hotelthree",
          "nodeType": "function",
          "props": Object {
            "facilities": Array [],
            "name": "hotelthree",
            "starRating": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
